controllers/dashboard-routes.js

```diff
@@ -0,0 +1,184 @@
+  // Retrieve the current post content
+  try {
+    const currPost = await Post.findOne({
+      where: {
+        id: req.params.id,
+      },
+      attributes: ["id", "title", "content", "created_at"],
+      include: [User],
+    });
+    res.render("editpost", {
+      post: currPost.get({ plain: true }),
+      loggedIn: req.session.loggedIn,
+    });
+  } catch (error) {
+    console.log(error);
+    res.render("editpost", {
+      error: "Something went wrong while retrieving the post",
+      loggedIn: req.session.loggedIn,
+    });
+  }
+});
+
+// get one of user's posts
+router.get("/post/:id", async (req, res) => {
+  try {
+    const post = await Post.findByPk(req.params.id, {
+      attributes: ["id", "title", "content", "created_at"],
+      include: [
+        {
+          model: Comment,
+          attributes: [
+            "id",
+            "comment_text",
+            "post_id",
+            "user_id",
+            "created_at",
+          ],
+          include: {
+            model: User,
+            attributes: ["username"],
+          },
+        },
+        {
+          model: User,
+          attributes: ["username"],
+        },
+      ],
+    });
+    // response
+    console.log(post.get({ plain: true }));
+    res.render("single-post", {
+      post: post.get({ plain: true }),
+      loggedIn: req.session.loggedIn,
+    });
+  } catch (error) {
+    console.log(error);
+    res.status(500).json("Error retrieving post data from database.");
+  }
+});
+
+// Find one of user's comments
+router.get("/comment/:id", async (req, res) => {
+  try {
+    const allComments = await Comment.findOne({
+      // where: {
+      //   user_id: req.session.user_id,
+      // },
+      attributes: ["id", "comment_text", "post_id", "user_id", "created_at"],
+      order: [["created_at", "DESC"]],
+      include: [
+        {
+          model: Post,
+          attributes: ["id", "title", "content", "created_at"],
+          include: {
+            model: User,
+            attributes: ["username"],
+          },
+        },
+        {
+          model: User,
+          attributes: ["username"],
+        },
+      ],
+    });
+    // response
+    res.redirect("/dashboard");
+  } catch (error) {
+    console.log(error);
+    res.status(500).json("Error retrieving comment data from database.");
+  }
+});
+
+// Edit (update) comment
+router.put("/comment/edit/:id", async (req, res) => {
+  try {
+    const data = await Comment.findByPk(req.params.id);
+    const comment = data.get({ plain: true });
+    // response
+    res.redirect("/dashboard");
+  } catch (error) {
+    console.log(error);
+    res.status(500).json("Error retrieving comment data from database.");
+  }
+});
+
+// Delete comment
+router.delete("/comment/:id", async (req, res) => {
+  try {
+    await Comment.destroy({
+      where: {
+        id: req.params.id,
+      },
+    });
+  } catch (error) {
+    console.error(error.message);
+  }
+});
+
```

Comment:
might be worht reviewing if all of these routes should be in this file...
There's a file breakdown on Slack that explains how a project should be structured and where certain routes go, please take a look but if you have any questions do let us know :-)